include/rocksdb/env.h

```diff
@@ -570,6 +570,26 @@ class SequentialFile {
   // SequentialFileWrapper too.
 };
 
+// A read IO request structure for use in MultiRead
+struct ReadRequest {
+  // File offset in bytes
+  uint64_t offset;
+
+  // Length to read in bytes
+  size_t len;
+
+  // A buffer that MultiRead()  can optionally place data in. It can
+  // ignore this and allocate its own buffer
+  char* scratch;
+
+  // Output parameter set by MultiRead() to point to the data buffer, and
+  // the number of valid bytes
+  Slice result;
+
+  // Status of read
+  Status status;
```

Comment:
Do we want a status per range, or one single status for everything. Imaging the underlying Env implementation consolidates it to one read request, it may not have a per request status.

Reply:
Having status per request will allow us to return partial success to the user (MultiGet returns status per key). Can't the underlying Env can simply replicate the failure status if it doesn't track per range?

Reply:
Do we want MultiGet() to handle partial success?

Reply:
That is a good question. The original MultiGet() has per-key status. I'm not sure if any users care about it or not. It doesn't cost us much, though, to handle partial success in terms of implementation complexity.

Reply:
The interface is good for me then. I'll defer to @riversand963 to finish the code review.